From ad62f7c3b7ba60182ae42d1573dae8fd57560ad7 Mon Sep 17 00:00:00 2001
From: Vlad Zahorodnii <vlad.zahorodnii@kde.org>
Date: Wed, 4 Aug 2021 20:28:51 +0300
Subject: [PATCH] Copy shape region to Deleted

Currently, the shape region is not copied to the Deleted. If either
SurfaceItemX11 or SurfaceItemXwayland needs to build quads for a shaped
window, it won't be able to do so properly because the corresponding
x11 window is long time gone.

If the shape region changes before the window is unmapped, you may still
see visual artifacts. Unfortunately, the only way to fix that bug is to
switch to wayland.

BUG: 440001
BUG: 438458
BUG: 435378


(cherry picked from commit e94fbcd853fe4101c9d82b4df0803c7b6e97c0d4)
---
 src/toplevel.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/toplevel.cpp b/src/toplevel.cpp
index 6d08ffe7b..d99ceaff0 100644
--- a/src/toplevel.cpp
+++ b/src/toplevel.cpp
@@ -126,6 +126,8 @@ void Toplevel::copyToDeleted(Toplevel* c)
     m_internalFBO = c->m_internalFBO;
     m_internalImage = c->m_internalImage;
     m_opacity = c->m_opacity;
+    m_shapeRegionIsValid = c->m_shapeRegionIsValid;
+    m_shapeRegion = c->m_shapeRegion;
 }
 
 // before being deleted, remove references to everything that's now
-- 
GitLab

